includes/admin/class-wc-rest-payments-orders-controller.php

```diff
@@ -199,4 +209,31 @@ public function create_customer( $request ) {
 			return new WP_Error( 'wcpay_server_error', __( 'Unexpected server error', 'woocommerce-payments' ), [ 'status' => 500 ] );
 		}
 	}
+
+	/**
+	 * Create a new in-person payment intent for the given order ID without confirming it.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function create_terminal_intent( $request ) {
+		// Do not process non-existing orders.
+		$order = wc_get_order( $request['order_id'] );
+		if ( false === $order ) {
+			return new WP_Error( 'wcpay_missing_order', __( 'Order not found', 'woocommerce-payments' ), [ 'status' => 404 ] );
+		}
+
+		try {
+			// Create a new intention.
+			$result = $this->gateway->create_intent( $order, [ Payment_Method::CARD_PRESENT ], 'manual' );
+			if ( is_wp_error( $result ) ) {
```

Comment:
This check is superfluous. [`rest_ensure_response`](https://developer.wordpress.org/reference/functions/rest_ensure_response/) will handle `WP_Error` for you.

Reply:
+1, the whole try-body can be simplified to `return rest_ensure_response( $this->gateway->create_intent( $order, [ Payment_Method::CARD_PRESENT ], 'manual' ) );`

Reply:
Ya'll got it 205f7b4 😄 

P.S. I like to keep method call and return statements on separate lines 😉

Reply:
Thanks!